infrastructure/src/test/java/org/corfudb/infrastructure/batchprocessor/BatchProcessorTest.java

```diff
@@ -0,0 +1,308 @@
+        verify(mockStreamLog).prefixTrim(sequence);
+    }
+
+    /**
+     * Test that the BatchProcessor successfully handles a WRITE request.
+     */
+    @Test
+    public void testWrite() {
+        LogData logData = getDefaultLogData(0L);
+        RequestMsg request = getRequestMsg(getBasicHeader(false, false),
+                getWriteLogRequestMsg(logData));
+
+        batchProcessor.addTask(BatchWriterOperation.Type.WRITE, request).join();
+        verify(mockStreamLog).append(0L, logData);
+    }
+
+    /**
+     * Test that the BatchProcessor successfully handles a RANGE_WRITE request.
+     */
+    @Test
+    public void testRangeWrite() {
+        final int numIter = 100;
+        List<LogData> entries = new ArrayList<>();
+        for (int x = 0; x < numIter; x++) {
+            LogData ld = getDefaultLogData(x);
+            entries.add(ld);
+        }
+
+        RequestMsg request = getRequestMsg(getBasicHeader(false, false),
+                getRangeWriteLogRequestMsg(entries));
+        batchProcessor.addTask(BatchWriterOperation.Type.RANGE_WRITE, request).join();
+        verify(mockStreamLog).append(entries);
+    }
+
+    /**
+     * Test that the BatchProcessor successfully handles a RESET request.
+     */
+    @Test
+    public void testReset() {
+        long epochWaterMark = 100L;
+        RequestMsg request = getRequestMsg(getBasicHeader(false, true),
+                getResetLogUnitRequestMsg(epochWaterMark));
+        batchProcessor.addTask(BatchWriterOperation.Type.RESET, request).join();
+        verify(mockStreamLog).reset();
+    }
+
+    /**
+     * Test that the BatchProcessor successfully handles a TAILS_QUERY request for LOG_TAIL.
+     */
+    @Test
+    public void testTailsQueryLogTail() {
+        RequestMsg request = getRequestMsg(getBasicHeader(false, false),
+                getTailRequestMsg(LogUnit.TailRequestMsg.Type.LOG_TAIL));
+
+        Object ret = batchProcessor.addTask(BatchWriterOperation.Type.TAILS_QUERY, request).join();
+        verify(mockStreamLog).getLogTail();
+        assertTrue(ret instanceof TailsResponse);
+    }
+
+    /**
+     * Test that the BatchProcessor successfully handles a TAILS_QUERY request for STREAMS_TAILS.
+     */
+    @Test
+    public void testTailsQueryStreamsTail() {
+        RequestMsg request = getRequestMsg(getBasicHeader(false, false),
+                getTailRequestMsg(LogUnit.TailRequestMsg.Type.STREAMS_TAILS));
+        List<UUID> list = request.getPayload()
+                .getTailRequest()
+                .getStreamList()
+                .stream()
+                .map(CorfuProtocolCommon::getUUID)
+                .collect(Collectors.toList());
+
+        when(mockStreamLog.getTails(any())).thenReturn(new TailsResponse(0L));
+        Object ret = batchProcessor.addTask(BatchWriterOperation.Type.TAILS_QUERY, request).join();
+        verify(mockStreamLog).getTails(list);
+        assertTrue(ret instanceof TailsResponse);
```

Comment:
I think we usually use this in unit tests:  `assertThat(ret).isInstanceOf(TailsResponse.class);`

Reply:
Done.